include/tensor.hpp

```diff
@@ -0,0 +1,30 @@
+#include <vector>
+#include <memory>
+
+namespace deepworks {
+
+using Shape = std::vector<size_t>;
```

Comment:
Shape can be `{-1, ...}`
I think it should be `class Shape : private std::vector<int>`

@l-bat

Reply:
I agree with @TolyaTalamanov, we can do like  https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/include/ngraph/shape.hpp#L30.
Add  operator<< for Shape